registry/remote/example_test.go

```diff
@@ -80,8 +106,40 @@ func TestMain(m *testing.M) {
 			w.WriteHeader(http.StatusCreated)
 		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleTag) && m == "PUT":
 			w.WriteHeader(http.StatusCreated)
-		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleRefereceDigest) && m == "PUT":
+		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleRefereceManifestDigest) && m == "PUT":
 			w.WriteHeader(http.StatusCreated)
+		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleSignatureManifestDescriptor.Digest) && m == "GET":
+			w.Header().Set("Content-Type", artifactspec.MediaTypeArtifactManifest)
+			w.Header().Set("Content-Digest", string(exampleSignatureManifestDescriptor.Digest))
+			w.Header().Set("Content-Length", strconv.Itoa(len(exampleSignatureManifest)))
+			if m == "GET" {
+				w.Write(exampleSignatureManifest)
+			}
+		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleSBoMManifestDescriptor.Digest) && m == "GET":
+			w.Header().Set("Content-Type", artifactspec.MediaTypeArtifactManifest)
+			w.Header().Set("Content-Digest", string(exampleSBoMManifestDescriptor.Digest))
+			w.Header().Set("Content-Length", strconv.Itoa(len(exampleSBoMManifest)))
+			if m == "GET" {
+				w.Write(exampleSBoMManifest)
+			}
+		case p == fmt.Sprintf("/v2/%s/_oras/artifacts/referrers", exampleRepositoryName):
+			q := r.URL.Query()
+			var referrers []artifactspec.Descriptor
+			switch q.Get("test") {
+			case "page1":
+				referrers = exampleReferrerDescriptors[1]
+			default:
+				referrers = exampleReferrerDescriptors[0]
+				w.Header().Set("Link", fmt.Sprintf(`<%s?n=1&test=page1>; rel="next"`, p))
+			}
+			result := struct {
+				Referrers []artifactspec.Descriptor `json:"referrers"`
+			}{
+				Referrers: referrers,
+			}
+			if err := json.NewEncoder(w).Encode(result); err != nil {
+				w.WriteHeader(http.StatusBadRequest)
```

Comment:
This would be a server-side error and should not happen. We can panic here.

Reply:
Changed to panic.